recipes/generate.py

```diff
@@ -81,9 +81,40 @@ def _setup_model(
 
         return model
 
+    def convert_prompt_to_tokens(
+        self,
+        prompt: Union[DictConfig, str],
+        chat_format: ChatFormat,
+        instruct_template: InstructTemplate,
+    ) -> List[Message]:
+        # Should only be chat-style prompt or instruct-style prompt
+        if chat_format and instruct_template:
```

Comment:
we should make `chat_format` and `instruct_template` optional - if both are required then this is inevitably raised, and it doesn't match the examples in your summary? also having both be none is a possible use case

Reply:
Ah sorry yeah this is what I had in mind anyways. Will update